app/models/bike.rb

```diff
@@ -149,6 +150,16 @@ def get_listing_order
     stock_photo_url.present? || public_images.present? ? t : t/100
   end
 
+  def creation_state; creation_states.first end
+
+  def creation_description; creation_state&.creation_description end
+  
+  def bulk_import; creation_state&.bulk_import end
+
+  def pos_kind; creation_state&.pos_kind end
```

Comment:
Suggestion / Question — [delegate](https://apidock.com/rails/Module/delegate) provides a succinct way to define this behavior / Should we prefer the longhand form?

Reply:
Yeah. This changed it from being `delegate` - I couldn't get the join working when it was `has_one`, and I couldn't get delegate working without a relation.

There is probably a way to fix either of those, and I'd be happy to use it! I'd prefer delegate for sure. I just didn't want to mess with it too much.

Reply:
Ah gotcha. Just checking, thanks 👍